smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/endpointsV2/ParameterGenerator.java

```diff
@@ -79,13 +79,15 @@ public String defaultAsCodeString() {
 
         switch (type.getValue()) {
             case "String":
+            case "string":
                 buffer += "\"" + paramNode.expectStringMember("default").getValue() + "\"";
                 break;
             case "Boolean":
+            case "boolean":
                 buffer += paramNode.expectBooleanMember("default").getValue() ? "true" : "false";
                 break;
             default:
-                // required by linter
+                throw new RuntimeException("Unhandled endpoint param type: " + type.getValue());
```

Comment:
it's okay to throw here since if you actually went in this branch the codegen would be `value ?? ,` which is a syntax error